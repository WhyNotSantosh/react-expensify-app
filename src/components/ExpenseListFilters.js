import React from "react";
import { connect } from "react-redux";
import { DateRangePicker } from "react-dates";
import {
  setFilterText,
  sortByDate,
  sortByAmount,
  setStartDate,
  setEndDate,
} from "../actions/filters";

export class ExpenseListFilters extends React.Component {
  state = {
    calendarFocused: null,
  };
  onDatesChange = ({ startDate, endDate }) => {
    this.props.setStartDate(startDate);
    this.props.setEndDate(endDate);
  };
  onFocusChange = (calendarFocused) => {
    this.setState({ calendarFocused });
  };
  onTextChange = (e) => {
    this.props.setFilterText(e.target.value);
  };
  onSortChange = (e) => {
    console.log(this.props.filters);
    if (e.target.value === "date") {
      this.props.sortByDate();
    } else if (e.target.value === "amount") {
      this.props.sortByAmount();
    }
  };
  render() {
    return (
      <div>
        <input
          type="text"
          value={this.props.filters.text}
          onChange={this.onTextChange}
        />
        <select value={this.props.filters.sortBy} onChange={this.onSortChange}>
          <option value="amount">Amount</option>
          <option value="date">Date</option>
        </select>
        <DateRangePicker
          startDate={this.props.filters.startDate}
          startDateId={"start"}
          endDate={this.props.filters.endDate}
          endDateId={"end"}
          onDatesChange={this.onDatesChange}
          focusedInput={this.state.calendarFocused}
          onFocusChange={this.onFocusChange}
          showClearDates={true}
          numberOfMonths={1}
          isOutsideRange={() => false}
        />
      </div>
    );
  }
}

const mapStateToProps = (state) => ({
  filters: state.filters,
});

const mapDispatchToProps = (dispatch) => ({
  setFilterText: (text) => dispatch(setFilterText(text)),
  sortByDate: () => dispatch(sortByDate()),
  sortByAmount: () => dispatch(sortByAmount()),
  setStartDate: (startDate) => dispatch(setStartDate(startDate)),
  setEndDate: (endDate) => dispatch(setEndDate(endDate)),
});

export default connect(mapStateToProps, mapDispatchToProps)(ExpenseListFilters);
