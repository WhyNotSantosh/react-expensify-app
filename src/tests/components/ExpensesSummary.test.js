import React from 'react';
import {shallow} from 'enzyme';
import { ExpenseSummary } from '../../components/ExpensesSummary';

test("should render ExpesesSummary with one expense", () => {
  const wrapper = shallow(<ExpenseSummary expenseCount={1} expensesTotal={235} />);
  expect(wrapper).toMatchSnapshot();
});

test("should render ExpesesSummary with multiple expenses", () => {
    const wrapper = shallow(<ExpenseSummary expenseCount={3} expensesTotal={4566985} />);
    expect(wrapper).toMatchSnapshot();
});