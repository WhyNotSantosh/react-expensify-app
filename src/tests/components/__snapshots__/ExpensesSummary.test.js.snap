// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render ExpesesSummary with multiple expenses 1`] = `
<div>
  <h1>
    View 
    3
     
    expenses
     Summary 
    $45,669.85
  </h1>
</div>
`;

exports[`should render ExpesesSummary with one expense 1`] = `
<div>
  <h1>
    View 
    1
     
    expense
     Summary 
    $2.35
  </h1>
</div>
`;
